gstd/src/exec.rs

```diff
@@ -16,19 +16,123 @@
+/// the current block timestamp, the current value of the gas available for execution.
+///
+/// The block height serves to identify a particular block.
+/// This information can be used to enable many scenarios like restricting or
+/// allowing of some functions until certain block height is reached.
+///
+/// The timestamp is the number of milliseconds elapsed since the Unix epoch.
+///
+/// Each message processing consumes gas, both on instructions execution and
+/// memory allocations. This function returns a value of the gas available for
+/// spending during current execution. Its use may help to avoid unexpected
+/// behaviors during the smart-contract execution in case of not enough gas
+/// available.
+/// 
+/// /// # Examples
+///
+/// ```
+/// use gstd::{exec, msg};
+///
+/// // Send a reply after the block height reaches the number 1000
+/// pub unsafe extern "C" fn handle() {
+///     if exec::block_height() >= 1000 {
+///         msg::reply(b"Block #1000 reached", 1_000_000, 0);
+///     }
+/// }
+/// ```
+/// ```
+/// use gstd::{exec, msg};
+///
+/// // Send a reply after the block timestamp reaches the February 22, 2022
+/// pub unsafe extern "C" fn handle() {
+///     if exec::block_timestamp() >= 1645488000000 {
+///         msg::reply(
+///             b"The current block is generated after February 22, 2022",
+///             1_000_000,
+///             0,
+///         );
+///     }
+/// }
+/// ```
+/// ```
+/// use gcore::exec;
+///
+/// // Perform work while `gas_available` is more than 1000
+/// pub unsafe extern "C" fn handle() {
+///     while exec::gas_available() > 1000 {
+///         // ...
+///     }
+/// }
+/// ```
 
 use crate::{ActorId, MessageId};
 pub use gcore::exec::{block_height, block_timestamp, gas_available};
 
+/// Pause the current message handling.
+///
+/// If the message handling needs to be paused, i.e. to wait for another
+/// execution to finish, this function should be used. [`wait`] finishes current
+/// message handle execution with a special result and puts the current message
+/// into the *waiting queue* to be awakened using the correspondent [`wake`]
+/// function later. All gas that hasn't yet been spent is attributed to the
+/// message in the *waiting queue*.
+///
+/// # Examples
+///
+/// ```
+/// use gstd::exec;
+///
+/// pub unsafe extern "C" fn handle() {
+///     // ...
+///     exec::wait();
+/// }
+/// ```
+
 pub fn wait() -> ! {
     gcore::exec::wait()
 }
 
+/// Resume previously paused message handling.
+///
+/// If a message has been paused using the [`wait`] function, then it is
+/// possible to continue its execution by calling this function. `waker_id`
+/// which specifies a particular message to be taken out of the *waiting queue*
+/// and put into the *processing queue*.
+///
+/// # Examples
+///
+/// ```
+/// use gstd::{exec, MessageId};
+///
+/// pub unsafe extern "C" fn handle() {
+///     // ...
+///     exec::wake(MessageId::default());
```

Comment:
```suggestion
///     ...
///     let msg_id = ... ;
///     exec::wake(msg_id);
```

Found it more correct